includes/admin/class-wc-admin-assets.php

```diff
@@ -417,6 +418,42 @@ public function admin_scripts() {
 				);
 			}
 		}
+
+		/**
+		 * Enqueue inline CSS for the admin menu.
+		 *
+		 * We register the top-level menu item as __( 'WooCommerce' ), which
+		 * causes the admin screen ID to be
+		 * `sanitize_title( __( 'WooCommerce' ) )`.
+		 *
+		 * This is still 'WooCommerce' for compatibility with extensions that
+		 * need to use screen IDs directly.  More info:
+		 *
+		 * - https://www.skyverge.com/blog/screen-id-checks-wordpress-submenu-pages/
+		 * - https://wpdirectory.net/search/01DSDNR9N8X79VRR1F01GZ3HVP
+		 * - https://github.com/ClassicPress-research/classic-commerce/pull/TODO
+		 *
+		 * To avoid showing WooCommerce in the admin menu, we override that
+		 * text with CSS, echoed inline so that the name can be translated.
+		 *
+		 * @since 1.0.0
+		 */
+		public function admin_menu_inline_css() {
+?>
+<style>
+ul#adminmenu > li#toplevel_page_woocommerce > a > div.wp-menu-name {
+	font-size: 0px;
+	color: rgba(0, 0, 0, 0);
+}
+ul#adminmenu > li#toplevel_page_woocommerce > a > div.wp-menu-name:after {
+	font-size: 14px;
+	color: #eee;
+	display: inline;
+	content: "<?php _e( 'Commerce', 'classic-commerce' ); ?>";
```

Comment:
If translations are not fully trusted then this may be a problem.  However there is not a built-in escaping function that is suitable here: https://stackoverflow.com/q/190396

Are translations echoed without escaping elsewhere in the code? (Probably so, if I had to guess)